app/models/data_set.rb

```diff
@@ -31,13 +31,19 @@
 class DataSet < ApplicationRecord
   belongs_to :cohort
   has_many :retweets
+  has_many :tweet_fetchers
 
   attr_readonly :index_name
   before_create :add_index_name
 
   def run_pipeline
     verify_index
-    ingest_data
+    schedule_ingest
+  end
+
+  def finish_when_ready
+    return unless complete?
```

Comment:
`finish` and `complete` are somewhat confusing here. I take it `finish` means something like aggregate, and `complete` is a adjective not a verb? Maybe `collected?` or something more descriptive? I understand that this is trying to stay generic because it's essentially event-driven without real events, so maybe it's fine. You use `finalize` instead of finish in a comment below, and for some reason that's clearer to me.

Reply:
👍 Thanks, this is exactly the sort of thing I can't see from inside my own head.